### PR TITLE
Implemented Binary Insertion Sort in Dart Language

### LINKED ISSUE #2125: Binary Insertion Sort
- [x] C
- [x] C++
- [x] C# - @Yashank18
- [ ] CoffeeScript
- [ ] Dart - @raksha009
- [ ] Go - @KamalDGRT
- [ ] Java - @raktim00
- [ ] JS - @sheetalssr
- [ ] Kotlin
- [ ] Php - @goyalmuskan
- [x] Python
- [ ] Readme - @goyalmuskan
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Jahnavi-Jonnalagadda**
Fixes #2125

#### Checklist:
- [ ] 4 space indentation.
- [ ] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [ ] Sample Input / Output is added at the end of file.
- [ ] Logic Documentation (Comments).
- [ ] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implemented Binary Insertion Sort in Dart
- 

#### Languages Used:
- Dart
- 

#### Files Added:
- Binary_Insertion_Sort.dart
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Jahnavi-Jonnalagadda** [comment]
@MastersAbh made the requested changes. Please review.

**Comment by raghav-dalmia** [comment]
Still your PR have 2 files. After completing all the changes, make sure to squash your commits.

File: Binary_Insertion_Sort.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/* Binary Insertion sort is a special type up of Insertion sort which uses binary search algorithm to find out the correct position 
+   of the inserted element in the array. */
+
+CODE:
```
Comment (MastersAbh) [inline Binary_Insertion_Sort.dart:4]: Please remove this line. I dont think it is required

File: Binary_Insertion_Sort.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			ar[j+1] = ar[j];
+			--j;
+		}
+		ar[j+1] = temp;
+	}
+}
+
+int binarySearch(List ar, int key, int start, int end)
+{
+    if(end-start <= 1)
```
Comment (MastersAbh) [inline Binary_Insertion_Sort.dart:29]: space after if

File: Binary_Insertion_Sort.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	}
+}
+
+main()
+{
+	List ar = [1, 5, 3, 4, 8, 6];
+	int n = ar.length;
+	insertionSort(ar);
+	print("Sorted Array: ");
+	for(int i = 0;i < n; i++)
```
Comment (MastersAbh) [inline Binary_Insertion_Sort.dart:61]: space after for

File: Binary_Insertion_Sort.dart [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		{
+			return start;
+		}
+	}
+	int mid = (start + end) ~/ 2;
+	if(ar[mid] < key)
+	{
+		return binarySearch(ar, key, mid, end);
+	}
+	else if(ar[mid] > key)
```
Comment (MastersAbh) [inline Binary_Insertion_Sort.dart:45]: space after if everywhere

**Comment by Jahnavi-Jonnalagadda** [comment]
Could you please help me out in squashing the commits?

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by MastersAbh** [comment]
https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

Please reach out on Slack if you still face issue

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by MastersAbh** [comment]
Hi @Jahnavi-Jonnalagadda  I dont see any update from your side, Pls let us know if you are working on this

**Comment by Jahnavi-Jonnalagadda** [comment]
I was unable to trace out how to squash commits. So could not proceed.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by MastersAbh** [comment]
Hi @Jahnavi-Jonnalagadda , If interested please reach out on slack channel, we will surely help you. If not, please close the PR.

**Comment by Jahnavi-Jonnalagadda** [comment]
Yeah, will reach out on the slack channel. Thank you.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by MastersAbh** [comment]
I dont see any update from @Jahnavi-Jonnalagadda ..Are you working on it? If not, we will close the PR

**Comment by MastersAbh** [comment]
Since we have no update, I am closing the PR.

**Pull request closed without merging**